backend/algorithms/create.py

```diff
@@ -81,37 +124,41 @@ def create_category(tokens):
     if re.match(r"^S$", tokens[0], flags=re.IGNORECASE):
         # School category
         return SchoolCategory(f"{tokens[0]} {tokens[1]}"), 1
+
     if re.match(r"^F$", tokens[0], flags=re.IGNORECASE):
         # Faculty category
         return FacultyCategory(f"{tokens[0]} {tokens[1]}"), 1
 
-    # TODO Levels (e.g. SPECIALISATIONS, PROGRAM)
+    if re.match(r"^[A-Z]{4}[0-9]{4}$", tokens[0], flags=re.IGNORECASE):
+        # Class category
+        return ClassCategory(tokens[0]), 0
+
+    # TODO: Levels (e.g. SPECIALISATIONS, PROGRAM)
+    # These don't have categories, do they need a category?
 
     # Did not match any category. Return None and assume only 1 token was consumed
     return None, 0
 
 
-def create_condition(tokens, course=None):
+def create_condition(tokens, course=None) -> CompositeCondition:
     """
     The main wrapper for make_condition so we don't get 2 returns.
     Given the parsed logical tokens list (assuming starting and ending bracket),
     and optionally a course for which this condition applies to,
-    Returns the condition
+    returns the condition
     """
     return make_condition(tokens, True, course)[0]
 
 
-def make_condition(tokens, first=False, course=None):
+def make_condition(tokens, first=False, course=None) -> CompositeCondition:
```

Comment:
this is also incorrect - we return a compositeCondition and how many tokens we consumed.